src/class-post-duplicator.php

```diff
@@ -0,0 +1,360 @@
+<?php
+/**
+ * Duplicate Post class to create copies.
+ *
+ * @package Duplicate_Post
+ * @since 4.0
+ */
+
+namespace Yoast\WP\Duplicate_Post;
+
+/**
+ * Represents the Post Duplicator class.
+ */
+class Post_Duplicator {
+
+	/**
+	 * Returns an array with the default option values.
+	 *
+	 * @return array The default options values.
+	 */
+	public function get_default_options() {
+		return [
+			'copy_title'             => true,
+			'copy_date'              => false,
+			'copy_status'            => false,
+			'copy_name'              => false,
+			'copy_excerpt'           => true,
+			'copy_content'           => true,
+			'copy_thumbnail'         => true,
+			'copy_template'          => true,
+			'copy_format'            => true,
+			'copy_author'            => false,
+			'copy_password'          => false,
+			'copy_attachments'       => false,
+			'copy_children'          => false,
+			'copy_comments'          => false,
+			'copy_menu_order'        => true,
+			'title_prefix'           => '',
+			'title_suffix'           => '',
+			'increase_menu_order_by' => null,
+			'parent_id'              => null,
+			'meta_excludelist'       => [],
+			'taxonomies_excludelist' => [],
+			'use_filters'            => true,
+		];
+	}
+
+	/**
+	 * Creates a copy of a post object, accordingly to an options array.
+	 *
+	 * @param \WP_Post $post    The original post object.
+	 * @param array    $options The options overriding the default ones.
+	 *
+	 * @return number|\WP_Error The copy ID, or a WP_Error object on failure.
```

Comment:
Shouldn't `number` be `int`?

Reply:
Yes! Copied from the legacy code, can't recall why it was like that, will fix there too